panel/src/panel/upload.js

```diff
@@ -74,12 +78,9 @@ export default (panel) => {
 			panel.dialog.close();
 
 			if (this.completed.length > 0) {
+				await this.emit("complete", this.completed);
 				await this.emit("done", this.completed);
-
-				if (panel.drawer.isOpen === false) {
```

Comment:
`Test:` Adding this also as a todo for myself to test if it still works properly inside drawers now this guard has been removed.

Reply:
You can use https://sandbox.test/panel/pages/other+uploads?tab=blocks to test this.